packages/aws-cdk-lib/aws-events/lib/rule.ts

```diff
@@ -320,6 +320,8 @@ export class Rule extends Resource implements IRule {
   private renderTargets() {
     if (this.targets.length === 0) {
       return undefined;
+    } else if (this.targets.length > 5) {
```

Comment:
What about adding the check inside the `validateRule` method instead?

Reply:
thanks makes sense. I moved it into `validateRule` method.